src/collections/Collections/Comparers/ArrayEqualityComparer.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace System.Collections.Generic;
+
+public sealed class ArrayEqualityComparer<T> : IEqualityComparer<T[]>
+{
+    private readonly IEqualityComparer<T>? comparer;
+
+    private IEqualityComparer<T> Comparer()
+        =>
+        comparer ?? EqualityComparer<T>.Default;
+
+    public ArrayEqualityComparer() { }
+
+    public ArrayEqualityComparer(IEqualityComparer<T>? comparer)
+        =>
+        this.comparer = comparer;
+
+    public static ArrayEqualityComparer<T> Default
+        =>
+        DefaultInstance.Value;
+
+    public bool Equals(T[]? x, T[]? y)
+    {
+        if (ReferenceEquals(x, y))
+        {
+            return true;
+        }
+
+        if (x is null || y is null)
+        {
+            return false;
+        }
+
+        if (x.Length != y.Length)
+        {
+            return false;
+        }
+
+        if (x.Length > 0)
```

Comment:
Is it better to use early return?

if (x.Length is not > 0)
{
   return true;
}

Reply:
Have fixed for both `Equals` and `GetHashCode` for all the comparers.
Also simplified making difference between null and empty collections (for performance purposes).